packages/eslint-plugin-office-addins/src/rules/call-sync-after-load.ts

```diff
@@ -0,0 +1,89 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import { Variable } from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import {
+  findPropertiesRead,
+  findReferences,
+  OfficeApiReference,
+} from "../utils";
+
+export = {
+  name: "call-sync-after-load",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"suggestion",
+    messages: {
+      callSyncAfterLoad:
+        "Call context.sync() after calling load on '{{name}}' property on the '{{loadValue}}' and before reading properties",
+    },
+    docs: {
+      description:
+        "Always call load on an object followed by a context.sync() before reading it or one of its properties.",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
+    let apiReferences: OfficeApiReference[] = [];
+
+    function findLoadBeforeSync(): void {
+      const needSync: Set<Variable> = new Set<Variable>();
+      const needLoadAndSync: Set<Variable> = new Set<Variable>();
+
+      apiReferences.forEach((apiReference) => {
+        const operation = apiReference.operation;
+        const reference = apiReference.reference;
+        const variable = reference.resolved;
+
+        if (operation === "Write" && variable) {
+          needLoadAndSync.add(variable);
+        }
+
+        if (operation === "Load" && variable) {
+          if (needLoadAndSync.has(variable)) {
+            needLoadAndSync.delete(variable);
+            needSync.add(variable);
+          }
+        }
+
+        if (operation === "Sync") {
+          needSync.clear();
+        }
+
+        const propertyName: string | undefined = findPropertiesRead(
+          reference.identifier.parent
+        );
+
+        if (
+          operation === "Read" &&
+          variable &&
+          (needSync.has(variable) || needLoadAndSync.has(variable))
+        ) {
```

Comment:
Tracking the sync of the variable is what the other rule does . . . this doesn't need to to that.